File: src/main/java/project/rpg/player/status/thing/Intelligence.java
package project.rpg.player.status.thing;

import org.bukkit.entity.Player;
import project.rpg.player.User;
import project.rpg.player.status.base.StatusBase;
import project.rpg.player.status.base.StatusName;

public class Intelligence extends StatusBase {   //마력

    public Intelligence() {
        super(StatusName.INTELLIGENCE);
    }

    @Override
    public void effect(Player player) {
        User.getPlayer(player).getMana().setMaxMana(this._value*10);
    }

}

File: src/main/java/project/rpg/player/status/objects/Intelligence.kt
package project.rpg.player.status.objects

import org.bukkit.entity.Player
import project.rpg.extensions.mana
import project.rpg.player.status.base.StatusBase
import project.rpg.player.status.base.StatusName

class Intelligence : StatusBase(StatusName.INTELLIGENCE) {
    override fun effect(player: Player?) {
        if (player != null) {
            player.mana.maxMana = value * 10
            player.mana.mana = value * 10
        }
    }
}